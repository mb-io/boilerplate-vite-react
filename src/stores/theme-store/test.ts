import { describe, it, expect } from 'vitest'
import { useThemeStore } from './theme-store'

describe('useThemeStore', () => {
  it('should have a default theme of "dark"', () => {
    const { theme } = useThemeStore.getState()
    expect(theme).toBe('dark')
    expect({ theme }).toMatchSnapshot()
  })

  it('should update the theme', () => {
    const { updateTheme } = useThemeStore.getState()
    updateTheme('light')
    const { theme } = useThemeStore.getState()
    expect(theme).toBe('light')
    expect({ theme }).toMatchSnapshot()
  })
})
