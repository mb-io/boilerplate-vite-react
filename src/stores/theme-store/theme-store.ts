import { create } from 'zustand'

import { Theme } from '@/hooks/use-theme/use-theme'

interface ThemeState {
  theme: Theme
  updateTheme: (theme: Theme) => void
}

export const useThemeStore = create<ThemeState>((set) => ({
  theme: 'dark', // default theme
  updateTheme: (theme) => set({ theme })
}))

export default useThemeStore
