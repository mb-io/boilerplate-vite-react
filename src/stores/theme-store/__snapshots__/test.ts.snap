// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useThemeStore > should have a default theme of "dark" 1`] = `
{
  "theme": "dark",
}
`;

exports[`useThemeStore > should update the theme 1`] = `
{
  "theme": "light",
}
`;
